linkml/generators/pydanticgen/array.py

```diff
@@ -394,30 +525,21 @@ def complex_dimensions(self, array: ArrayExpression) -> SlotResult:
                 # equivalent to labeled shape
                 return self.parameterized_dimensions(array)
             else:
-                raise ValueError(
-                    "if exact_number_dimensions is provided, it must be greater than the parameterized dimensions"
-                )
+                # invalid - raise exception
+                ArrayValidator.array_consistent_n_dimensions(array)
```

Comment:
Since `validate` is called at the beginning of `make`, this shouldn't trigger, right? I guess, unless someone calls `complex_dimensions` without calling `make` first. I don't mind having it here, but maybe we should note that or add `# pragma: no cover` since it should not be reachable.

Reply:
oh good catch. ya the validator was intended to remove the need for all these check cases. I guess it's here just for logical completeness. something feels wrong about just falling through but you're right it shouldn't be reachable. maybe one thing is that even though they're not part of the intended API via the `make` method, they are public methods, so it would be rare but possible?

if we took this out we fall through to the error message on L473, which is more generic.

so i'm thinking we either make these private methods and take this out, or another thing i could do is run each of the validation failure tests against the relevant method if we're gonna leave em as public. also fine with taking this out np.

Reply:
I like making these private, taking out these errors, and maybe leaving a comment saying that the missing branch should not occur because validation happens first or something like that.

Reply:
I cant think of a circumstance where it would be preferable to call the generation method over the `make` method. Lets make em private.

Im gonna assume the methods in the Validator class are clear enough as documentation for what legs of the if/else clauses should be impossible and just remove them entirely